moonshot/src/api/api_session.py

```diff
@@ -1,172 +1,136 @@
+        api_load_runner(runner_id).database_instance, runner_id, context_strategy
+    )
 
 
-def api_update_context_strategy(session_id: str, context_strategy_name: str) -> None:
+def api_update_prompt_template(runner_id: str, prompt_template: str) -> None:
     """
-    Updates the context strategy for a specific session.
+    Updates the prompt template for a specific runner.
 
-    This API endpoint calls the `Session.update_context_strategy` method to update the context strategy
-    associated with the specified session ID. It allows clients to modify the context strategy for a session,
-    enabling dynamic changes in how context is managed within the session.
+    This function updates the prompt template for a specific runner identified by the given runner_id. It calls the
+    `Session.update_prompt_template` method with the runner's database instance,
+    runner_id, and the new prompt_template.
 
     Args:
-        session_id (str): The unique identifier of the session for which the context strategy is to be updated.
-        context_strategy_name (str): The new context strategy name to be assigned to the session.
+        runner_id (str): The ID of the runner for which the prompt template needs to be updated.
+        prompt_template (str): The new prompt template to be set for the runner.
 
     Returns:
-        None: This method does not return a value but updates the context strategy for the specified session.
+        None
     """
-    Session.update_context_strategy(session_id, context_strategy_name)
+    Session.update_prompt_template(
+        api_load_runner(runner_id).database_instance, runner_id, prompt_template
+    )
 
 
-def api_update_prompt_template(session_id: str, prompt_template_name: str) -> None:
+def api_delete_session(runner_id: str) -> None:
     """
-    Updates the prompt template for a specific session.
+    Deletes the session for a specific runner.
 
-    This API endpoint calls the `Session.update_prompt_template` method to update the prompt template
-    associated with the specified session ID. It allows clients to modify the prompt template for a session,
-    enabling dynamic changes in the prompts generated within the session.
+    This function deletes the session for the runner identified by the given runner_id. It calls the
+    `Session.delete_session` method with the runner's database instance.
 
     Args:
-        session_id (str): The unique identifier of the session for which the prompt template is to be updated.
-        prompt_template_name (str): The new prompt template name to be assigned to the session.
+        runner_id (str): The ID of the runner for which the session needs to be deleted.
 
     Returns:
-        None: This method does not return a value but updates the prompt template for the specified session.
+        None
     """
-    Session.update_prompt_template(session_id, prompt_template_name)
+    Session.delete_session(api_load_runner(runner_id).database_instance)
```

Comment:
i assume this delete the session which is the row in the session table? the session table goes with it or just the record

Reply:
this will delete the session_metadata_table and all the endpoint tables. when the user uses the same runner to do red teaming, the tables will then be created again